cmd/nsm-init/nsm-init.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"net"
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/ligato/networkservicemesh/pkg/nsm/apis/nsmconnect"
+	"google.golang.org/grpc"
+)
+
+const (
+	// clientConnectionTimeout defines time the client waits for establishing connection with the server
+	clientConnectionTimeout = time.Second * 60
+)
+
+var (
+	clientSocket = flag.String("nsm-socket", "/var/lib/networkservicemesh/nsm.ligato.io.sock", "Location of NSM process client access socket")
```

Comment:
Make the location of the socket a const we can pickup from somewhere else.

Reply:
Please remember that it is external executable file, the only way you can pass the socket location is via flags passed to it at start time. It is a normal practice to give an option to override the default behaviour.

Reply:
Ah, that makes sense. Thanks!

Reply:
I agree with this sentiment, set the default as a constant and make it overrideable. Just make sure that there are no cycles in package imports if you take this path.